powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/v3/ActivationServiceBehavior.java

```diff
@@ -1452,26 +1452,14 @@ public RemoveActivationResponse removeActivation(String activationId, String ext
      * @return Response with confirmation of removal.
      */
     public RemoveActivationResponse removeActivation(@NotNull ActivationRecordEntity activation, String externalUserId, boolean revokeRecoveryCodes) {
+        logger.info("Processing activation removal, activation ID: {}", activation.getActivationId());
         activation.setActivationStatus(io.getlime.security.powerauth.app.server.database.model.ActivationStatus.REMOVED);
-        if (revokeRecoveryCodes) {
-            final RecoveryCodeRepository recoveryCodeRepository = repositoryCatalogue.getRecoveryCodeRepository();
-            final List<RecoveryCodeEntity> recoveryCodeEntities = recoveryCodeRepository.findAllByActivationId(activation.getActivationId());
-            final Date now = new Date();
-            for (RecoveryCodeEntity recoveryCode : recoveryCodeEntities) {
-                // revoke only codes that are not yet revoked, to avoid messing up with timestamp
-                if (!RecoveryCodeStatus.REVOKED.equals(recoveryCode.getStatus())) {
-                    recoveryCode.setStatus(RecoveryCodeStatus.REVOKED);
-                    recoveryCode.setTimestampLastChange(now);
-                    // Change status of PUKs with status VALID to INVALID
-                    for (RecoveryPukEntity puk : recoveryCode.getRecoveryPuks()) {
-                        if (RecoveryPukStatus.VALID.equals(puk.getStatus())) {
-                            puk.setStatus(RecoveryPukStatus.INVALID);
-                            puk.setTimestampLastChange(now);
-                        }
-                    }
-                    recoveryCodeRepository.save(recoveryCode);
-                }
-            }
+        // Recovery codes are revoked in case revocation is requested, or always when the activation is in CREATED or PENDING_COMMIT state
+        if (revokeRecoveryCodes ||
+                (activation.getActivationStatus() == ActivationStatus.CREATED ||
```

Comment:
are we using == style or ActivationStatus.CREATED.equals(activation.getActivationStatus())?

Reply:
Good catch, we typically use equals...

Reply:
There is no point in using equals for enums, enumerations are null safe. @banterCZ Please add your comment.

Reply:
Exactly
> Testing equality of an enum value with equals is perfectly valid because an enum is an Object and every Java developer knows "==" should not be used to compare the content of an Object. At the same time, using "==" on enums:
>
> - provides the same expected comparison (content) as equals
> - is more null-safe than equals()
> - provides compile-time (static) checking rather than runtime checking
> For these reasons, use of "==" should be preferred to equals.

https://rules.sonarsource.com/java/RSPEC-4551